src/EcommerceStatsBundle/Service/StatsTableService.php

```diff
@@ -0,0 +1,225 @@
+            $condition = '';
+            if (count($conditionList) > 0) {
+                $condition = 'WHERE ('.implode(') AND (', $conditionList).')';
+            }
+
+            $blockQuery = str_replace(['[{sColumnName}]', '[{sCondition}]'], [$dateQueryPart, $condition], $baseQuery);
+            $groupFields = explode(',', $group->fieldGroups);
+            $realGroupFields = array_filter(array_map('trim', $groupFields));
+
+            $this->addBlock($blocks, $group->fieldName, $blockQuery, $realGroupFields, $params);
+        }
+
+        return new StatsTableDataModel(
+            $blocks,
+            $this->getColumnNames($blocks),
+            $showDiffColumn,
+            $this->getMaxGroupColumnCount($blocks)
+        );
+    }
+
+    private function getBaseConditions(TdbPkgShopStatisticGroup $group, string $startDate, string $endDate, string $portalId): array
+    {
+        $baseConditionList = [];
+        $params = [];
+
+        $baseConditionList[] = $this->connection->quoteIdentifier(str_replace('`', '', $group->fieldDateRestrictionField)).' >= :from';
+        $params[':from'] = $startDate;
+
+        $baseConditionList[] = $this->connection->quoteIdentifier(str_replace('`', '', $group->fieldDateRestrictionField)).' <= :to';
+        $params[':to'] = $endDate;
+
+        if ('' !== $group->fieldPortalRestrictionField && '' !== $portalId) {
+            $baseConditionList[] = $this->connection->quoteIdentifier(str_replace('`', '', $group->fieldPortalRestrictionField)).' = :portalId';
+            $params[':portalId'] = $portalId;
+        }
+
+        return [$baseConditionList, $params];
+    }
+
+    /**
+     * @return Generator<TdbPkgShopStatisticGroup>
+     */
+    private function fetchStatistics(): Generator
+    {
+        $groups = TdbPkgShopStatisticGroupList::GetList();
+        while ($group = $groups->Next()) {
+            yield $group;
+        }
+    }
+
+    private function getDateQueryPart(string $dateGroupType): string
+    {
+        return self::DATE_QUERY_PARTS[$dateGroupType]
+            ?? self::DATE_QUERY_PARTS[self::DATA_GROUP_TYPE_DAY];
+    }
+
+    /**
+     * add a new block to the list.
+     *
+     * @param array<string, StatsGroupDataModel> $blocks
+     * @param string[]                           $subGroups
+     * @param array<string, string>              $params
+     */
+    protected function addBlock(
+        array &$blocks,
+        string $blockName,
+        string $query,
+        array $subGroups = [],
+        array $params = []
+    ): void {
+        if (false === array_key_exists($blockName, $blocks)) {
+            $ecommerceStatsGroup = new StatsGroupDataModel();
+            $ecommerceStatsGroup->init($blockName);
+            $blocks[$blockName] = $ecommerceStatsGroup;
+        }
+
+        foreach ($this->fetchRows($query, $params) as $dataRow) {
+            if (!array_key_exists('sColumnName', $dataRow) || !array_key_exists('dColumnValue', $dataRow)) {
```

Comment:
Where do these names (sColumnName, dColumnValue) in Hungarian notation come from?

Reply:
These are part of the queries that are currently stored in the database by the to-be-deprecated stats bundle.

Reply:
I documented the expected column names and where they are coming from in the readme file now.